yogstation/code/game/objects/items/bandage.dm

```diff
@@ -0,0 +1,147 @@
+		healing_limb.bandaged = FALSE
+		used = TRUE
+
+
+/obj/item/medical/bandage/proc/fall_off(mob/living/carbon/human/H, obj/item/bodypart/L)
+	if (L.bandaged)
+		to_chat(H, "You loosen the bandage around [L.name] and let it fall to the floor.")
+		name = "used [src.name]"
+		desc = "Bloodied and crusted, these bandages have clearly been used and aren't fit for much anymore. Seems as if they were wrapped around someone's [L.name] last."
+		color = "red"
+		loc = H.loc
+		L.bandaged = FALSE
+		used = TRUE
+
+/obj/item/medical/bandage/proc/apply(mob/living/user, mob/tar, obj/item/bodypart/lt)
+	if (!ishuman(user))
+		to_chat(user, "<span class='warning'>You don't have the dexterity to use this!</span>")
+		return 0
+
+	if (ishuman(tar))
+		if (!lt.bandaged)
+			if (user == tar)
+				user.visible_message("<span class='notice'>[user] begins winding [src] about their [lt.name]..</span>", "<span class='notice'>You begin winding [src] around your [lt.name]..</span>")
+			else
+				user.visible_message("<span class='notice'>[user] begins winding [src] about [tar]'s [lt.name]..</span>", "<span class='notice'>You begin winding [src] around [tar]'s [lt.name]..</span>")
+
+			if (do_after(user, 50, target = tar))
+				if(!user.canUnEquip(src))
+					return 0
+				healing_limb = lt
+				lt.bandaged = src
+				moveToNullspace(src)
+				user.visible_message("[user] has applied [src] successfully.", "You have applied [src] successfully.")
+				return 1
+			else
+				user.visible_message("<span class='warning'>[user] stops applying [src] to [tar].</span>", "<span class='warning'>You stop applying [src] to [tar].</span>")
+				return 0
+		else
+			to_chat(user, "[tar] is already bandaged for the moment.")
+			return 0
+	else
+		to_chat(user, "This doesn't look like it'll work.")
+		return 0
+
+/obj/item/medical/bandage/proc/wash(obj/O, mob/user)
+	if (src.used)
+		to_chat(user, "You clean [src] fastidiously washing away as much of the detritus and residue as you can. The bandage can probably be used again now.")
+		name = "reused bandages"
+		desc = "Whatever quality these bandages once were, there's no sign of it any more. Not like the wounds you put this stuff over care, though."
+		healamount = src.healamount * 0.85
+		duration = src.duration * 1.15
+		activefor = 1
+		color = 0
+		used = FALSE
+	else
+		to_chat(user, "There's no real need to wash this - it's perfectly clean!")
+
+/obj/item/medical/bandage/attack(mob/living/carbon/human/T, mob/living/carbon/human/U)
+	if (used)
+		to_chat(U, "These bandages have already been used. They're worthless as they are. Maybe if they had the blood washed out of them with running water?")
+		return
+
+	var/obj/item/bodypart/O =  T.get_bodypart(check_zone(U.zone_selected))
+
+	if (O.status == ORGAN_ROBOTIC)
+		to_chat(U, "You don't have time to explain why there's no time to explain why you can't bandage this very obviously robotic limb.")
+		return
+
+	if (!O.bandaged)
+		apply(U, T, O)
+		return
+	else if (O.bandaged)
```

Comment:
good code

Reply:
Like actually or should I fix something?  I just ported it so I don't even know.

Reply:
Yes, fix it.